frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -393,101 +392,93 @@ function removeForestClientFromList(index: number) {
-                                    :options="applicationRoleOptions"
-                                    optionLabel="role_name"
-                                    optionValue="role_id"
-                                    :modelValue="field.value"
-                                    placeholder="Choose an option"
-                                    class="w-100"
-                                    style="width: 100% !important"
-                                    v-bind="field.value"
-                                    @update:modelValue="handleChange"
-                                    @change="resetForestClientNumberData()"
-                                    :class="{
-                                        'is-invalid': errors.role_id,
-                                    }"
-                                >
-                                </Dropdown>
-                            </Field>
-                            <ErrorMessage
-                                class="invalid-feedback"
-                                name="role_id"
-                            >
-                            </ErrorMessage>
-                        </div>
-                    </div>
-
-                    <div v-if="isAbstractRoleSelected()">
-                        <div class="row mb-0">
-                            <div class="px-0 col-md-6">
+                </StepContainer>
+
+                <StepContainer
+                    title="Add user roles"
+                    subtitle="Enter a specific role for this user"
+                    :divider="isAbstractRoleSelected()"
+                >
+                    <label>Assign a role to the user</label>
+
+                    <Field
+                        name="role_id"
+                        aria-label="Role Select"
+                        v-slot="{ field, handleChange }"
+                        v-model="formData.role_id"
+                    >
+                        <Dropdown
+                            :options="applicationRoleOptions"
+                            optionLabel="role_name"
+                            optionValue="role_id"
+                            :modelValue="field.value"
+                            placeholder="Choose an option"
+                            class="w-100 custom-input"
+                            style="width: 100% !important"
+                            v-bind="field.value"
+                            @update:modelValue="handleChange"
+                            @change="resetForestClientNumberData()"
+                            :class="{
+                                'is-invalid': errors.role_id,
+                            }"
+                        >
+                        </Dropdown>
+                    </Field>
+                    <ErrorMessage class="invalid-feedback" name="role_id">
+                    </ErrorMessage>
+                </StepContainer>
+
+                <div v-if="isAbstractRoleSelected()">
```

Comment:
this v-if can be added to the StepContainer

Reply:
Good suggestion, I have added this as well. Thank you.